homeassistant/components/arcticspa/config_flow.py

```diff
@@ -0,0 +1,97 @@
+"""Config flow for Arctic Spa integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from pyarcticspas import Spa
+from pyarcticspas.error import SpaHTTPException, TooManyRequestsError, UnauthorizedError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): str,
+    }
+)
+
+
+def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+
+    device = Spa(data[CONF_API_KEY])
+
+    try:
+        _ = device.status()
+    except UnauthorizedError:
+        raise InvalidAuth
+    except TooManyRequestsError:
+        raise TooManyRequests
+    except SpaHTTPException as ex:
+        _LOGGER.exception("Unexpected error code %d", ex.code)
+        raise CannotConnect
+
+    # Return info that you want to store in the config entry.
+    return {
+        "name": f"API-{device.id[:8]}",
+        "id": device.id,
+        CONF_API_KEY: data[CONF_API_KEY],
+    }
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle the config flow for Arctic Spa."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            try:
+                info = validate_input(self.hass, user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except TooManyRequests:
+                errors["base"] = "too_many_requests"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(info["id"])
```

Comment:
This ID is now unique for every hottub right?

Reply:
Yes.

device.id is hashed from the API key && every API key == one hot tub => every ID == one hot tub.

Reply:
That also leaves me with a question: can an API key change?

Reply:
In short: I don't think so.

Currently, the API is in beta testing at the manufacturer. This means that you request API keys through e-mail and you get a non-expiring key from them. You can't revoke a key yourself and the developers are not keen on revoking or changing keys (or regular feedback discussions for that matter).

If a key changes, the unique ID changes too, so you'll have to delete your old key and install your new key in Home Assistant.